torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -749,8 +749,10 @@ def state_dict_type(
         Args:
             module (torch.nn.Module): Root module.
             state_dict_type (StateDictType): the desired ``state_dict_type`` to set.
-            state_dict_config (Optional[StateDictConfig]): the configuration for the
-                target ``state_dict_type``.
+            state_dict_config (Optional[StateDictConfig]): the model state_dict
```

Comment:
<img width="729" alt="Screenshot 2023-06-02 at 8 36 29 AM" src="https://github.com/pytorch/pytorch/assets/31054793/3be55d68-561c-465d-9546-0ffc50267cd5">

nit: I think we should use double backtick for "state_dict" or just call it "state dict" without the underscore.